unity-renderer/Assets/UnityGLTF/Scripts/GLTFComponent.cs

```diff
@@ -339,29 +283,80 @@ public IEnumerator LoadAssetCoroutine(Settings settings)
                             sceneImporter = null;
                         }
-
-                        loader = null;
                     }
-
-                    alreadyLoadedAsset = true;
-
-                    CoroutineStarter.Stop(loadingRoutine);
-                    loadingRoutine = null;
-
-                    if ( state == State.COMPLETED )
-                        OnFinishedLoadingAsset?.Invoke();
-                    else
-                        OnFailedLoadingAsset?.Invoke(new Exception($"GLTF state finished as: {state}"));
-
-                    Destroy(loadingPlaceholder);
-                    Destroy(this);
+                    
+                    if (!isDestroyed)
+                    {
+                        if (!token.IsCancellationRequested)
+                        {
+                            if ( state == State.COMPLETED)
+                                OnFinishedLoadingAsset?.Invoke();
+                            else
+                                OnFailedLoadingAsset?.Invoke(new Exception($"GLTF state finished as: {state}"));
+                        }
+                    
+                        CleanUp();
+                        Destroy(loadingPlaceholder);
+                        Destroy(this);
+                        isDestroyed = true;
+                    }
                 }
             }
             else
             {
                 Debug.Log("couldn't load GLTF because url is empty");
             }
         }
+        private void OverrideShaders()
+        {
+            Renderer[] renderers = gameObject.GetComponentsInChildren<Renderer>();
+
+            foreach (Renderer renderer in renderers)
+            {
+                renderer.sharedMaterial.shader = shaderOverride;
+            }
+        }
+
+        private void SetupSceneImporter(Settings settings, string id, ILoader loader)
+        {
+            sceneImporter = new GLTFSceneImporter(
+                id,
+                GLTFUri,
+                loader,
+                throttlingCounter
+            );
+
+            if (sceneImporter.CreatedObject != null)
+            {
+                Destroy(sceneImporter.CreatedObject);
+            }
+
+            sceneImporter.SceneParent = gameObject.transform;
+            sceneImporter.Collider = Collider;
+            sceneImporter.maximumLod = MaximumLod;
+            sceneImporter.useMaterialTransition = UseVisualFeedback;
+            sceneImporter.CustomShaderName = shaderOverride ? shaderOverride.name : null;
+            sceneImporter.LoadingTextureMaterial = LoadingTextureMaterial;
+            sceneImporter.initialVisibility = initialVisibility;
+            sceneImporter.addMaterialsToPersistentCaching = addMaterialsToPersistentCaching;
+
+            sceneImporter.forceGPUOnlyMesh = settings.forceGPUOnlyMesh
+                                             && DataStore.i.featureFlags.flags.Get().IsFeatureEnabled(FeatureFlag.GPU_ONLY_MESHES);
```

Comment:
This will cause tech debt when we try to extract the importer to other package. We should be injecting the options from outside instead of having the GLTFComponent know about how the outside world works.